### PR TITLE
update delete and patch should return ROS

### LINKED ISSUE #196: update and delete should return proper objects
create and get return RecursiveObjectStruct ... update and delete return RestClient::Response ... not good ...

### PR DESCRIPTION
**Pull request by moolitayer**
Fix #196 

Public methods update_x delete_x and patch_x should all return RecursiveOpenStruct like get_x and watches.

I did not add the `:as => :raw` options in these cases because the response is small and limited so I do not expect to hit performance problems there. Anyway since the `:as` is an additional option it can be added in a minor version.

### CODE REVIEW

**Comment by abonas** [comment]
Can u please clarify in the commit message what does format mean here

**Comment by moolitayer** [comment]
@abonas updated please take a look

**Comment by moolitayer** [comment]
@grosser @kirs please @cben please review

**Comment by abonas** [comment]
can you please explain why a delete method should return ROS?

**Comment by moolitayer** [comment]
Why return anything?

Because the API server gives you the last state of the object (it will be an object in deletion) and we are throwing that away.

Why ROS?

Compatibility with other methods mostly.
I mentioned in the first comment why I did not add the :as option

File: test/test_replication_controller.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Replication Controller entity tests
 class TestReplicationController < MiniTest::Test
-  def test_get_from_json_v1
+  def tes_get_from_json_v1
```
Comment (cben) [inline test/test_replication_controller.rb]: typo (test -> tes)

**Comment by cben** [comment]
@abonas any objections?

**Pull request merged**